File: December/Day1/Day1/1704_Determine_if_String_Halves_Are_Alike.java
class Solution {

    private int countVowels(String s){
        int count =0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='a' || s.charAt(i)=='e' || s.charAt(i)=='i' || s.charAt(i)=='o' || s.charAt(i)=='u' || s.charAt(i)=='A' || s.charAt(i)=='E' || s.charAt(i)=='I' || s.charAt(i)=='O' || s.charAt(i)=='U')
                count++;
        }
        return count;
    }
    public boolean halvesAreAlike(String s) {
        int n =s.length();
        String s1 = s.substring(0,n/2);
        String s2 = s.substring(n/2);

        return countVowels(s1)==countVowels(s2);
    }
}

File: December/Day1/1704_Determine_if_String_Halves_Are_Alike.cpp
class Solution {
public:
bool isVowel(char c){
        if(c=='a' || c=='A' || c=='e' || c=='E' || c=='i' || c=='I' || c=='o' || c=='O' || c=='u' || c== 'U')
            return true;
        return false;
    }
    bool halvesAreAlike(string s) {
        int l=s.length();
        int c1=0,c2=0;
        for(int i=0; i<l/2; ++i){
            if(isVowel(s[i])) ++c1;
        }
        for(int i=l/2; i<l; ++i){
            if(isVowel(s[i])) ++c2;
        }
        return c1==c2;
    }
};
